### DISCUSSION TITLE
How to tell if in debug mode?

### DISCUSSION THREAD

**Discussion by moonman239** [pallets/flask#4850] [Q&A]
I am working on a SaaS application that accepts payments through Stripe. I have to use different keys for Stripe's API, depending on whether I'm testing my application or the application is in production.
For example, the application should use the API key "stripe-test-1234" in my development environment, and "stripe-prod-5678" in my production environment.

Because of this, I'd like to have some code that looks like this:

```
if app.mode == "DEBUG":
    stripe.api_key = "stripe-test-1234"
else:
    stripe.api_key = "stripe-prod-5678"
```

Is there a way to check if the application has been run with the --debug option? If so, how?

Thanks for your assistance.

**Comment by davidism** [comment]
Do that in the test setup, not in your app setup. That way you know the code is only set in tests.

**Reply by moonman239** [reply]
Good idea, but this doesn't really answer the titular question.

**Reply by davidism** [reply]
`app.debug`

**Reply by moonman239** [reply]
So:

```
if (app.debug):
    # debug mode
else:
    # production mode
```
?
